### PR TITLE
Added centos based caddy server.

### PR DESCRIPTION
**Pull request by mohammedzee1000**
https://caddyserver.com/

### CODE REVIEW

**Comment by mohammedzee1000** [comment]
ping @kbsingh @jperrin review please

File: caddy_server/centos7/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+## Caddy Server Dockerfile
+https://caddyserver.com
+
+This is a dockerfile for Caddy Server. Everything is loaded into `/srv` directory. The Caddyfile is put into `/srv/config/Caddyfile`
+
+By default, content present inside `/srv/public` will be served over web on the port 2015. Ports 80, 443, 8080 and 8443 are also exposed. Mount your content here with appropriate permissions.
```
Comment (kbsingh) [inline caddy_server/centos7/README.md]: Can you change this to be more in line with the default CentOS linux target for /var/www/html/

File: caddy_server/centos7/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+MAINTAINER Mohammed Zeeshan Ahmed <moahmed@redhat.com>
+LABEL VERSION="0.9.5" ARCHITECTURE="amd64"
+ARG plugins=git
+RUN yum -y update && yum clean all
+RUN yum -y install openssh-client git && yum clean all
+RUN useradd -u 1001 -g 0 -d /srv caddy
+
+RUN curl --silent --show-error --fail --location \
+      --header "Accept: application/tar+gzip, application/x-gzip, application/octet-stream" -o - \
+      "https://caddyserver.com/download/build?os=linux&arch=amd64&features=${plugins}" \
```
Comment (kbsingh) [inline caddy_server/centos7/Dockerfile]: is this the best packaging we can find for it ? 

also how are you going to track updates / releases for it ?

Comment (mohammedzee1000) [inline caddy_server/centos7/Dockerfile]: Yes, the only other place you can get it is github releases, and that will probably make it harder to enable plugins with. Although we could track the the git releases for updates

File: caddy_server/centos7/README.md (hunk 1 above)
Comment (mohammedzee1000) [inline caddy_server/centos7/README.md]: Done

File: caddy_server/centos7/Dockerfile (hunk 2 above)
Comment (mohammedzee1000) [inline caddy_server/centos7/Dockerfile]: I also subscribed myself to recieve noti for new releases over email

File: caddy_server/centos7/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+## Caddy Server Dockerfile
+https://caddyserver.com
+
+This is a dockerfile for Caddy Server. Everything is loaded into `/srv` directory. The Caddyfile is put into `/etc/caddy/Caddyfile`
```
Comment (vpavlin) [inline caddy_server/centos7/README.md]: Is `/srv` correct here? I see `WORKDER /var/www/html`

Comment (mohammedzee1000) [inline caddy_server/centos7/README.md]: @vpavlin thanks will update

**Comment by vpavlin** [comment]
I'd say it looks good...Let's build it and test it

**Comment by kbsingh** [comment]
need the corrosponding entry in the container-index as well

**Comment by mohammedzee1000** [comment]
Adding

**Comment by mohammedzee1000** [comment]
Container has been built from my branch available on registry.centos.org as caddyserver/caddyserver:latest

@vpavlin @kbsingh please go ahead and test it.

**Comment by mohammedzee1000** [comment]
Updated

**Comment by kbsingh** [comment]
Thanks for all the work on getting this done

**Pull request merged**